thebeetoken/beenest-web
# Fix precedence of dates for hotel listings

## Description
Noticed in the context of #34 that searched dates were overriding fixed dates for conference hotel listings, sometimes causing invalid dates to be presented as a booking option. This reverses the precedence order of these properties to display fixed dates for these listings as expected.

## How to Test
1. Make a search query with check-in/out dates specified
    1. Make sure query includes some conference listing results!
2. Click over to a conference listing
    1. Verify that dates shown in booking request card correspond to the *conference*
3. Go back to search results
4. Click over to a non-conference listing
    1. Verify that dates shown in booking request card correspond to the search query

Which devices did you test on?

## REVIEWERS:
Check against these principles:

### High level
Does this code need to be written?
What are the alternatives?
Will this implementation become a support issue?
How much error margin does this solution have?

### Code
* Does the code follow industry standards?
JS: https://github.com/airbnb/javascript
React: https://github.com/airbnb/javascript/tree/master/react
https://github.com/vasanthk/react-bits
Documentation headers: http://usejsdoc.org/index.html

* Is there duplicated code? Can it be refactored into a shared method?
* Is the code consistent with our project?
* Are there unit tests? Do they test the states?
* Is the person refactoring another developer's code? If possible, did the original developer approve?

### Variables/Naming:
* Would the variable type led to future edge cases?
* Are the variable naming clear? Would the value contain something other than what the name describes.

### Security
* Can this be hacked or abused by the user?

## Further Work
This is going to be an instant merge conflict with #34, so I'll be resolving conflicts. Will it be here? Will it be there? We'll find out!
